### PR TITLE
feat(ui): add Kord Breach season planner with personal modifiers

### PR DESCRIPTION
**Pull request by DysektAI**
## Summary

- Add Kord Breach seasonal character modifier planning tool with 25 modifiers (16 positive, 9 negative, 6 hardcore)
- Point budget system: positive modifiers cost points, negative modifiers grant points, plan is valid when total >= 0
- Pinia store with localStorage persistence and stale-ID hydration guard (`normalizeSelection`)
- Page component with SEO meta tags, modifier grouping, validation alert, and reset
- Reusable ModifierCard component with accessibility attributes (`aria-pressed`)
- Navigation integration in drawer
- i18n keys in `en.json` (Crowdin handles propagation)
- Store tests (8 suites), component tests (5 suites), and page tests (4 suites) — 21 tests total, all passing

## Test plan

- [x] Store tests pass (`useSeasonPlanner.test.ts` — 8 suites)
- [x] Component tests pass (`ModifierCard.test.ts` — 5 suites)
- [x] Page tests pass (`season-planner.page.test.ts` — 4 suites)
- [x] Lint clean (zero warnings)
- [x] Typecheck clean
- [ ] CodeRabbit review
- [ ] CodeQL scan
- [ ] CI pipeline (lint, typecheck, test, validate, workers, supabase)

## Summary by CodeRabbit

* **New Features**
  * Added a Season Planner page accessible from the navigation drawer.
  * Users can browse/select positive, negative, and hardcore modifiers; live point total updates and selection disables incompatible options.
  * Added alerts for invalid point totals and modifier conflicts, plus a reset action.
  * Added localized SEO metadata for the Season Planner page.
* **Localization**
  * Added English translations for all Season Planner text, alerts, and the seasonal modifier catalog.
* **Tests**
  * Added unit/component tests covering modifier card states, planner page interactions, and store logic (totals, validation, persistence, and reset).

___

## **CodeAnt-AI Description**
Add a Kord Breach season planner for building valid character modifier plans

### What Changed
- Added a Season Planner page linked from the navigation, with positive, negative, and hardcore seasonal rules
- Users can select personal modifiers, balance point costs and rewards, view full descriptions, and reset their plan
- Incompatible modifiers are blocked from selection and persisted conflicts show a dedicated warning
- Plans persist between visits while outdated, duplicate, hardcore, and conflicting selections are removed
- Modifier names and descriptions support localization with English fallback text
- Added coverage for modifier selection, accessibility, localization, validation, conflicts, reset behavior, persistence cleanup, and SEO metadata

### Impact
`✅ Clearer season modifier planning`
`✅ Fewer invalid modifier combinations`
`✅ Persistent plans across visits`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

<details open><summary><h3>Greptile Summary</h3></summary>

This PR adds a Kord Breach Season Planner feature: a point-budget planner where users pick personal modifiers (16 positive, 9 negative) that cost or grant points, plus 6 display-only hardcore rules. The plan is valid when the running total is ≥ 0 and no incompatible pair is selected.

- New Pinia store (`useSeasonPlanner`) with user-scoped localStorage persistence, a custom serializer that rejects foreign-user or malformed data, bidirectional conflict checking, and a `normalizeSelection` hydration guard called on page mount.
- `ModifierCard` component, `season-planner.vue` page with SEO meta and validation alerts, drawer navigation entry, full type definitions, and 21 tests (store, component, page) covering edge cases including multi-user isolation, malformed storage, and singular/plural point messaging.
- `docs/SYSTEMS.md` extended with the required system 15 section (summary, Mermaid diagram, behaviour details, file list, invariants).
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

- This is a self-contained, client-only feature with no server-side changes, migrations, or auth impact. The store's persistence layer and conflict logic are well-guarded and thoroughly tested.
- All new code is additive, the user-scoped serializer prevents cross-user data leakage, and the 21-test suite covers edge cases (malformed storage, multi-user isolation, conflict normalization). The only finding is a naming convention inconsistency in the storage key that carries no runtime risk.
- app/utils/storageKeys.ts — the new seasonPlanner key is the only entry without the shared STORAGE_PREFIX.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| app/stores/useSeasonPlanner.ts | New Pinia store with user-scoped localStorage persistence, robust input sanitisation via normalizeSelectedModifiers, bidirectional conflict checking in conflictsWithSelection, and a prepareForCurrentUser guard that clears state on user change. Logic is clean and well-tested. |
| app/utils/storageKeys.ts | Adds seasonPlanner key using a bare string 'season-planner-v1' instead of the ${STORAGE_PREFIX}… pattern used by every other key in this file — the only key without the tarkovtracker- namespace prefix. |
| app/pages/season-planner.vue | New page component with SEO meta, conflict/validity alerts, and three modifier columns. Calls normalizeSelection on mount. isModifierDisabled correctly delegates to the store's conflictsWithSelection action. |
| app/features/season-planner/ModifierCard.vue | Accessible button card with aria-pressed, disabled propagation, and i18n fallback for name/description. Native disabled prevents click emission so the toggle guard is browser-safe. |
| app/data/seasonal-modifiers.ts | Static modifier catalogue with createHardcoreModifier helper and satisfies-checked array. Incompatibility is declared symmetrically on both sides of each pair, which is consistent with the bidirectional check in conflictsWithSelection. |
| docs/SYSTEMS.md | New section 15 documents the season planner system with a Mermaid flow diagram, behaviour details, file list, and invariants. Satisfies AGENTS.md requirements. |

</details>

<details open><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant LS as localStorage<br/>(season-planner-v1)
    participant S as useSeasonPlannerStore
    participant P as season-planner.vue
    participant C as ModifierCard

    P->>S: useSeasonPlannerStore()
    S->>LS: deserialize (SEASON_PLANNER_SERIALIZER)
    LS-->>S: validate _userId + normalizeSelectedModifiers
    P->>S: onMounted → normalizeSelection()
    S->>S: prepareForCurrentUser() + strip conflicts

    P->>S: personalModifiers / allModifiers (getters)
    S-->>P: filtered modifier lists
    P->>C: :modifier :selected :disabled
    C-->>P: "@toggle"

    P->>S: toggleModifier(id)
    S->>S: prepareForCurrentUser()
    S->>S: conflictsWithSelection(id, selectedIds)?
    S->>S: splice / push selectedModifiers
    S->>LS: serialize → serializeUserScopedStorage
```
</details>

<sub>Reviews (3): Last reviewed commit: ["Merge branch &#39;main&#39; into kord-breach"](https://github.com/tarkovtracker-org/tarkovtracker/commit/7fc1f924369a02269bedbf2d6f97948b23cf4da4) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=45813845)</sub>

**Context used:**

- Context used - AGENTS.md ([source](https://app.greptile.com/dysektai/github/tarkovtracker-org/tarkovtracker/-/custom-context?memory=635c24fb-71e5-43a4-9979-74405b725e95))

### CODE REVIEW

**Comment by DysektAI** [comment]
The modifier names/descriptions in `seasonal-modifiers.ts` serve as English fallback values. The UI resolves them via `t(`page.season_planner.modifiers.${id}.name`, modifier.name)` in both `ModifierCard.vue` and `season-planner.vue`, with all 36 locale keys added to `en.json`. This matches the project's established pattern (`t('key', 'English fallback')` per AGENTS.md) — the data file is the source of truth for fallbacks, not for localization keys themselves.

File: app/features/drawer/DrawerLinks.vue [hunk 1]
```diff
@@ -19,6 +19,12 @@
         :is-collapsed="props.isCollapsed"
         :badge="t('page.storyline.wip_badge')"
       ></DrawerItem>
+      <DrawerItem
+        icon="i-heroicons-sparkles"
+        locale-key="season_planner"
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/features/drawer/DrawerLinks.vue]: Addressed in 7ea9072. Added navigation_drawer.season_planner to the English source locale and corrected DrawerLinks to use the full key consumed by DrawerItem.

File: app/pages/season-planner.vue [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        <div class="mb-6 flex flex-col gap-4 sm:flex-row sm:items-center sm:justify-between">
+          <div class="flex items-center gap-3">
+            <span
+              class="bg-primary-500/15 border-primary-500/25 inline-flex h-10 w-10 shrink-0 items-center justify-center rounded-lg border"
+            >
+              <UIcon name="i-heroicons-sparkles" class="text-primary-300 h-6 w-6" />
+            </span>
+            <div>
+              <h1 class="text-surface-100 text-2xl font-bold tracking-tight">
+                {{ t('page.season_planner.title', 'Season Planner') }}
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/pages/season-planner.vue]: Checked against repository localization policy. This repository explicitly requires editing only app/locales/en.json; non-English locale exports are Crowdin-owned and missing keys intentionally fall back to English. The new source keys and explicit page fallbacks are present in en.json, so adding generated translations here would violate the repository contract. No non-English files were changed.

File: app/pages/season-planner.vue [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    description: () =>
+      t(
+        'page.season_planner.seo_description',
+        'Plan your Escape from Tarkov Kord Breach seasonal character modifiers and point balance.'
+      ),
+  });
+  const plannerStore = useSeasonPlannerStore();
+  onMounted(() => {
+    plannerStore.normalizeSelection();
+  });
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/pages/season-planner.vue]: Addressed in 7ea9072. Added a Season planner section to docs/SYSTEMS.md with a summary, mermaid flow, step-by-step behavior, implementing files, and invariants covering point totals, conflicts, hydration normalization, and persistence safety.

File: app/stores/useSeasonPlanner.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      }
+      return selectedIds.some((selectedId) => {
+        const selected = this.personalModifiers.find((candidate) => candidate.id === selectedId);
+        return selected?.incompatibleWith?.includes(id);
+      });
+    },
+  },
+  persist: {
+    key: 'season-planner-v1',
+  },
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/stores/useSeasonPlanner.ts]: Addressed in 7ea9072. Season planner persistence keeps the original key for compatibility, but its serializer stores a user-scoped envelope and rejects mismatched user IDs during hydration. The store also clears an in-memory selection when the authenticated user changes. Added tests for cross-user hydration and user switching.

File: app/locales/en.json [hunk 5]
```diff
@@ -1648,6 +1648,168 @@
...
+      "seo_title": "Season Planner - Kord Breach",
+      "seo_description": "Plan your Escape from Tarkov Kord Breach seasonal character modifiers and point balance.",
+      "positive_modifiers": "Positive Modifiers",
+      "negative_modifiers": "Negative Modifiers",
+      "hardcore_modifiers": "Hardcore Rules",
+      "hardcore_description": "Global rules that apply to all seasonal characters. They don't affect your points but define the season's challenge.",
+      "total_points": "Total Points",
+      "reset": "Reset Plan",
+      "invalid_total_title": "Invalid Point Total",
+      "points_needed": "Add {points} more points to make this plan valid.",
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/locales/en.json]: Addressed in 7ea9072. This duplicate pluralization finding is covered by the singular/plural points_needed messages and the page passing the absolute missing-point count to t. The page test covers the singular case.

[the remark this answers is not in the document]
Comment (DysektAI) [reply app/locales/en.json]: Addressed in 7ea9072. points_needed now has singular/plural pipe variants, and the page passes the absolute missing-point count to t. Added a page test covering the singular count.

**Comment by DysektAI** [comment]
Review reconciliation: all seven previously unresolved threads were addressed with code or repository-policy evidence, replied to, and resolved. The implementation now user-scopes persisted planner selections, sanitizes rehydrated state, handles bidirectional incompatibility checks, uses plural-aware point messaging, supplies the drawer locale key, and documents the system in docs/SYSTEMS.md. Local verification passed: full Vitest (274 files, 2,667 tests), typecheck, lint, format, Fallow, i18n, OpenAPI, and CI=true production build. The current hosted CI run is green; the only failed status still displayed is the stale pre-fix SonarCloud Code Analysis run, while the current SonarCloud check is green.

**Comment by DysektAI** [comment]
Automated-review note for head 16dbce9611d3fcedcb4de86b415ffed1c15a002a: CodeRabbit completed as rate-limited, so it produced no fresh automated review for this head. I re-read all prior reviews, summary comments, and inline threads; all applicable findings are implemented and the unresolved-thread count is 0. Direct self-review found no new issues, and the hosted code, security, test, typecheck, build, lint, and validation checks are green. No findings are being deferred.

**Comment by DysektAI** [comment]
## Branch refreshed against `main` (head `ebe1b100`)

- Merged `origin/main` (`4d8128f8`, 55 commits behind). The only conflict was `docs/SYSTEMS.md`: main added sections 15–17, so the Season planner section is renumbered to **§18** and the TOC now lists 17 and 18.
- Committed the pending `.some(...)` guard simplification in `useSeasonPlanner.toggleModifier` (`36c11e31`) — this also clears the one open SonarCloud issue on this PR (`typescript:S7754`).
- Local validation at `ebe1b100` (clean worktree): `pnpm run validate:changes --base origin/main` (lint, prettier, typecheck, workflow tests, full Vitest 334 files / 3,926 tests, i18n, systems drift) ✅ · `pnpm run lint:fallow --base origin/main` ✅ · focused season-planner + drawer suites 70/70 ✅.

Season 1 runs 2026-08-03 → 2026-12-07, so the planner is still in-season. Whether it should leave draft now is a product call.
